lottie/src/main/java/com/airbnb/lottie/network/NetworkFetcher.java

```diff
@@ -80,69 +68,41 @@ private LottieComposition fetchFromCache() {
     return null;
   }
 
+  @NonNull
   @WorkerThread
-  private LottieResult<LottieComposition> fetchFromNetwork() {
-    try {
-      return fetchFromNetworkInternal();
-    } catch (IOException e) {
-      return new LottieResult<>(e);
-    }
-  }
-
-  @WorkerThread
-  private LottieResult<LottieComposition> fetchFromNetworkInternal() throws IOException {
+  private LottieResult<LottieComposition> fetchFromNetwork(@NonNull String url, @Nullable String cacheKey) {
     Logger.debug("Fetching " + url);
 
-
-    HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
-    connection.setRequestMethod("GET");
-
+    LottieFetchResult fetchResult = null;
     try {
-      connection.connect();
-
-      if (connection.getErrorStream() != null || connection.getResponseCode() != HttpURLConnection.HTTP_OK) {
-        String error = getErrorFromConnection(connection);
-        return new LottieResult<LottieComposition>(new IllegalArgumentException("Unable to fetch " + url + ". Failed with " + connection.getResponseCode() + "\n" + error));
+      fetchResult = fetcher.fetchSync(url);
+      if (fetchResult.isSuccessful()) {
+        InputStream inputStream = fetchResult.bodyByteStream();
+        String contentType = fetchResult.contentType();
+        LottieResult<LottieComposition> result = fromInputStream(url, inputStream, contentType, cacheKey);
```

Comment:
You marked bodyByteStream as nullable but pass it into this not-null method. Does your IDE not warn you about this?

Reply:
marked it NonNull

Reply:
marked it NonNull